components/ui/Badge.tsx

```diff
@@ -13,16 +13,40 @@ export default function Badge({
   style,
   ...rest
 }: BadgeProps) {
-  const variantStyle = variant === "outline" ? styles.outline : styles.default
-  const customTextStyle =
-    variant === "outline" ? styles.outlineText : styles.defaultText
+  const variantStyle: Record<BadgeVariants, ViewStyle> = {
+    default: {
+      backgroundColor: theme.colors.black,
+      paddingVertical: 1,
+      paddingHorizontal: 6,
+    },
+    outline: {
+      paddingVertical: 4,
+      paddingHorizontal: 8,
+      borderColor: theme.colors.black,
+      borderWidth: 1,
+    },
+    primary: {
+      backgroundColor: theme.colors.white,
+      paddingVertical: 1,
+      paddingHorizontal: 8,
+    },
+  }
```

Comment:
I think it would be cleaner and consistent to keep the styles in the StyleSheet.create object and call them in the here

```suggestion
  const variantStyle = {
    default: styles.default,
    outline: styles.outline,
    primary: styles.primary
  }
  ...
  const styles = StyleSheet.create({
  ...
  default: {
      backgroundColor: theme.colors.black,
      paddingVertical: 1,
      paddingHorizontal: 6,
    },
    outline: {
      paddingVertical: 4,
      paddingHorizontal: 8,
      borderColor: theme.colors.black,
      borderWidth: 1,
    },
    primary: {
      backgroundColor: theme.colors.white,
      paddingVertical: 1,
      paddingHorizontal: 8,
    }

  }
```